### ISSUE TITLE
Get error in training with resume=True

### ISSUE THREAD

**Issue by mrljwlm** [ultralytics/ultralytics#834]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

the error occur here in line 541: [ultralytics/yolo/engine/trainer.py](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/yolo/engine/trainer.py#:~:text=def%20resume_training(,total%20epochs%27))

```
529.     def resume_training(self, ckpt):
530.         if ckpt is None:
531.             return
532.         best_fitness = 0.0
533.         start_epoch = ckpt['epoch'] + 1
534.         if ckpt['optimizer'] is not None:
535.             self.optimizer.load_state_dict(ckpt['optimizer'])  # optimizer
536.             best_fitness = ckpt['best_fitness']
537.         if self.ema and ckpt.get('ema'):
538.             self.ema.ema.load_state_dict(ckpt['ema'].float().state_dict())  # EMA
539.             self.ema.updates = ckpt['updates']
540.         if self.resume:
541.             assert start_epoch > 0, \
542.                 f'{self.args.model} training to {self.epochs} epochs is finished, nothing to resume.\n' \
543.                 f"Start a new training without --resume, i.e. 'yolo task=... mode=train model={self.args.model}'"
544.             LOGGER.info(
545.                 f'Resuming training from {self.args.model} from epoch {start_epoch + 1} to {self.epochs} total epochs')
```

 in line 533: start_epoch = ckpt['epoch'] + 1 
**however ckpt['epoch'] is always -1** when my training model was strip_optimized in [ultralytics/yolo/utils/torch_utils.py](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/yolo/utils/torch_utils.py#:~:text=%5D%20%3D%20None-,x%5B%27epoch%27%5D%20%3D%20%2D1,-x%5B%27model) in line 335
```
329.     x = torch.load(f, map_location=torch.device('cpu'))
330.     args = {**DEFAULT_CFG_DICT, **x['train_args']}  # combine model args with default args, preferring model args
331.     if x.get('ema'):
332.         x['model'] = x['ema']  # replace model with ema
333.     for k in 'optimizer', 'best_fitness', 'ema', 'updates':  # keys
334.         x[k] = None
335.     x['epoch'] = -1
336.     x['model'].half()  # to FP16
337.     for p in x['model'].parameters():
338.         p.requires_grad = False
339.     x['train_args'] = {k: v for k, v in args.items() if k in DEFAULT_CFG_KEYS}  # strip non-default keys
340.     # x['model'].args = x['train_args']
341.     torch.save(x, s or f)
342.     mb = os.path.getsize(s or f) / 1E6  # filesize
343.     LOGGER.info(f"Optimizer stripped from {f},{f' saved as {s},' if s else ''} {mb:.1f}MB")
```

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by blueclowd** [comment]
> @mrljwlm 👋 Hello! Thanks for asking about **resuming training**. YOLO 🚀 Learning Rate (LR) [schedulers](https://github.com/ultralytics/yolov5/blob/aa542ce6a65658ff931fee9bbab77c0145c152f0/train.py#L175-L181) follow predefined LR curves for the fixed number of `epochs` defined at training start (`default=100`), and are designed to fall to a minimum LR on the final epoch for best training results. For this reason you can not modify the number of epochs once training has started.
> 
> Once a model has it's optimizer stripped it can naturally no longer be resumed from. You can always use it to start new trainings though:
> 
> ```
> yolo train model=path/to/best.pt
> ```

Hello, I have questions

Is there a way to disable the optimiser strip if I really want to resume the training from this weight?
